fix: add module name

Jappie3 opened the pull request:
This MR adds the module name in zephyr/module.yml, as is stated in the [official documentation](https://docs.zephyrproject.org/latest/develop/modules.html#module-name):

> Each Zephyr module is given a name by which it can be referred to in the build system.
> The name should be specified in the zephyr/module.yml file. This will ensure the module name is not changeable through user-defined directory names or west manifest files.

tomi-font reviewed (changes requested):
Thanks for the fix. There should be a single commit and no merge commits as part of normal PRs (so basically the second commit shouldn't be there).

Jappie3 commented:
done

tomi-font reviewed (changes requested):
Please add some of the PR description to the commit message for context. Also, the commit title should rather follow the structure `<area touched>: <what is done>` so e.g. `zephyr/module.yml: add module name`

Jappie3 commented:
something like this?

Jappie3 committed:
zephyr/module.yml: add module name

The module name is now explicitly set in module.yml, this ensures that
building with e.g. cmake does not fail when the directory name is
different (default name if unset is directory name).

tomi-font reviewed (approved):
Yes!
